tools/src/main/java/org/apache/kafka/tools/ConsumerPerformance.java

```diff
@@ -378,8 +402,10 @@ public Optional<Pattern> include() {
                     : Optional.empty();
         }
 
-        public long numMessages() {
-            return options.valueOf(numMessagesOpt);
+        public long numRecords() {
+            return options.has(numMessagesOpt)
+                    ? options.valueOf(numMessagesOpt)
+                    : options.valueOf(numRecordsOpt);
```

Comment:
I think we should also add a warning message to notify users when they use the deprecated `messages` option.